internal/rbd/rbd_util.go

```diff
@@ -1934,6 +1946,29 @@ func (rv *rbdVolume) setVolumeMetadata(parameters map[string]string) error {
 	return nil
 }
 
+// unsetVolumeMetadata unset PV/PVC/PVCNamespace metadata on RBD image.
+func (rv *rbdVolume) unsetVolumeMetadata() error {
+	list, err := rv.ListMetadata()
+	if err != nil {
+		return fmt.Errorf("failed to list metadata image %s/%s: %w", rv.Pool, rv.RbdImageName, err)
+	}
+
+	for k, v := range list {
+		for _, key := range k8s.GetVolumeMetadataKeys() {
+			if strings.Compare(k, key) != 0 {
```

Comment:
why `strings.Compare` and not just `k == key`?

Reply:
I was over thinking may be :-)